### PR TITLE
Further improve TS declarations

### PR DESCRIPTION
**Pull request by willeastcott**
Further improvements to the TS declarations:

* Document properties in `Application` class in-place.
* Improve `shape` API by removing `property` tags and hiding private properties.
* Fix type of `KeyboardEvent` with `globalThis` object.
* Fix various other typos/grammatical problems.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/input/mouse-event.js [hunk 1]
```diff
@@ -53,7 +38,19 @@ class MouseEvent {
         }
 
         if (coords) {
+            /**
+             * The x co-ordinate of the mouse pointer relative to the element {@link Mouse} is
```
Comment (slimbuck) [inline src/input/mouse-event.js:42]: coordinate?

File: src/input/touch-event.js [hunk 2]
```diff
@@ -31,56 +31,88 @@ function getTouchTargetCoords(touch) {
...
- * @property {Touch[]} changedTouches A list of touches that have changed since the last event.
- * @property {TouchEvent} event - The original browser TouchEvent.
  */
 class TouchEvent {
     /**
      * Create a new TouchEvent instance. It is created from an existing browser event.
      *
      * @param {TouchDevice} device - The source device of the touch events.
-     * @param {TouchEvent} event - The original browser TouchEvent.
+     * @param {globalThis.TouchEvent} event - The original browser TouchEvent.
```
Comment (slimbuck) [inline src/input/touch-event.js:89]: what is globalThis?

**Review by slimbuck — APPROVED** [review]
Approve with small comment.

File: src/input/touch-event.js (hunk 2 above)
Comment (willeastcott) [reply src/input/touch-event.js:89]: See here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis

It distinguishes the DOM's `TouchEvent` from `pc.TouchEvent`, say.

File: src/input/mouse-event.js (hunk 1 above)
Comment (willeastcott) [reply src/input/mouse-event.js]: Good point. Fixed.

**Pull request merged**